### PR TITLE
UUID base all

### PR DESCRIPTION
**Pull request by InnocentBug**
# Description

According to @brili all nodes (including subobjects) have a UUID, and a few other attributes.
So, I added those attributes to the UUIDBaseNode and have all nodes inherit from it.

## Known Issues
This might be better done with refactoring, and considering how BaseNode and UUIDBaseNode now serve a similar function.

## Checklist

- [x] My name is on the list of contributors (`CONTRIBUTORS.md`) in the pull request source branch.
- [ ] I have updated the documentation to reflect my changes.
Documentation isn't up to date with this.

### CODE REVIEW

**Comment by InnocentBug** [comment]
@nh916 does this cover the purpose of #143 ?

**Comment by nh916** [comment]
> @nh916 does this cover the purpose of #143 ?

it does not, because file still needs to have attributes such as `name` `notes` and the other primary base attributes

**Review by nh916 — APPROVED** [review]

File: src/cript/nodes/core.py [hunk 1]
```diff
@@ -69,10 +69,18 @@ def __setattr__(self, key, value):
...
         for kwarg in kwargs:
             if kwarg not in tolerated_extra_json:
-                try:
-                    getattr(self._json_attrs, kwarg)
-                except KeyError:
-                    raise CRIPTExtraJsonAttributes(self.node_type, kwarg)
+                if kwarg.endswith("_count"):
+                    try:
+                        possible_array = getattr(self._json_attrs.kwarg[: -len("_count")])
+                        if not isinstance(possible_array, list):
```
[the remark this answers is not in the document]
Comment (InnocentBug) [reply src/cript/nodes/core.py:75]: yeah, agreed.
It isn't pretty, just a retro fit to match the repsonse of the API.
As anticipation of our integration testing.

**Comment by InnocentBug** [comment]
> > @nh916 does this cover the purpose of #143 ?
> 
> it does not, because file still needs to have attributes such as `name` `notes` and the other primary base attributes

Ok, alright, I just wanted to bring it to your attention to avoid double the work.

**Comment by nh916** [comment]
@InnocentBug is there any way we can get trunk to delete the branch after it was merged?

**Comment by InnocentBug** [comment]
I thought that was the default. 
In my other repos it does that.

**Comment by nh916** [comment]
> I thought that was the default. In my other repos it does that.

hmm, then maybe it was just this one. I'll keep an eye out

**Pull request merged**